<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Cheminformatics Drug Similarity • SPcheminfo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Cheminformatics Drug Similarity">
<meta property="og:description" content="SPcheminfo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-CCMXLYXWYV"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-CCMXLYXWYV');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPcheminfo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SPcheminfo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/systemPipeR/SPcheminfo/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Cheminformatics Drug Similarity</h1>
                        <h4 data-toc-skip class="author">Author:
FirstName LastName</h4>
            
            <h4 data-toc-skip class="date">Last update: 02 October,
2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/SPcheminfo/blob/HEAD/vignettes/SPcheminfo.Rmd" class="external-link"><code>vignettes/SPcheminfo.Rmd</code></a></small>
      <div class="hidden name"><code>SPcheminfo.Rmd</code></div>

    </div>

    
    
<!--
Config css and r style
-->
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="about-the-template">About the template<a class="anchor" aria-label="anchor" href="#about-the-template"></a>
</h2>
<p>This section provides general description and how to use this
cheminformatics workflow. In the actual analysis report, this section is
usually <strong>removed</strong>.</p>
<p>This cheminformatics workflow template is based on the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/ChemmineR/inst/doc/ChemmineR.html" class="external-link">ChemmineR</a>
package and should be downloaded from <a href="https://www.bioconductor.org/packages/release/bioc/html/ChemmineR.html" class="external-link">Bioconductor</a>
before running the workflow. This template is a workflow that does:</p>
<ol style="list-style-type: decimal">
<li>Loads molecules from SDF format.</li>
<li>Help users visualize specified molecules.</li>
<li>Convert molecules to AP and FP format.</li>
<li>Compute the similarity and distance between molecules</li>
<li>Plot the distance matrix.</li>
</ol>
<p>There is no other command-line software required in this workflow.
All are written in R (<code>Linewise</code>) steps.</p>
<p><img src="results/plotwf_spcheminfo.jpg"></p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Users want to provide here background information about the design of
their cheminformatics project.</p>
<p>This report describes the analysis of a cheminformatics project
studying drug …</p>
<div class="section level3">
<h3 id="experimental-design">Experimental design<a class="anchor" aria-label="anchor" href="#experimental-design"></a>
</h3>
<p>Typically, users want to specify here all information relevant for
the analysis of their Cheminformatics study. This includes detailed
descriptions of files, experimental design, reference genome, gene
annotations, etc.</p>
</div>
</div>
<div class="section level2">
<h2 id="workflow-environment">Workflow environment<a class="anchor" aria-label="anchor" href="#workflow-environment"></a>
</h2>
<p><em><code>systemPipeR</code></em> workflows can be designed and built
from start to finish with a single command, importing from an R Markdown
file or stepwise in interactive mode from the R console.</p>
<p>This tutorial will demonstrate how to build the workflow in an
interactive mode, appending each step. The workflow is constructed by
connecting each step via <code>appendStep</code> method. Each
<code>SYSargsList</code> instance contains instructions needed for
processing a set of input files with a specific command-line or R
software and the paths to the corresponding outfiles generated by a
particular tool/step.</p>
<p>To create a Workflow within <em><code>systemPipeR</code></em>, we can
start by defining an empty container and checking the directory
structure:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://systempipe.org/" class="external-link">systemPipeR</a></span><span class="op">)</span></span>
<span><span class="va">sal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SPRproject.html" class="external-link">SPRproject</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">sal</span></span></code></pre></div>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<p>This is an empty template that contains only one demo step. Refer to
our <a href="https://systempipe.org/sp/spr/spr_run/" class="external-link">website</a> for how
to add more steps. If you prefer a more enriched template, <a href="https://systempipe.org/sp/spr/templates/" class="external-link">read this page</a> for
other pre-configured templates.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu">crayon</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/crayon/man/crayon.html" class="external-link">blue</a></span><span class="op">$</span><span class="fu">bold</span><span class="op">(</span><span class="st">"To use this workflow, following R packages are expected:\n"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"'ChemmineR"</span>, <span class="st">"ggplot2"</span>, <span class="st">"tibble"</span>, <span class="st">"readr"</span>, <span class="st">"ggpubr"</span>, <span class="st">"gplots'\n"</span><span class="op">)</span>,</span>
<span>    sep <span class="op">=</span> <span class="st">"', '"</span><span class="op">)</span></span>
<span><span class="co">### pre-end</span></span>
<span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://systempipe.org/" class="external-link">systemPipeR</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ChemmineR</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"load_packages"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-sdf-dataset">Load SDF dataset<a class="anchor" aria-label="anchor" href="#load-sdf-dataset"></a>
</h3>
<p>Molecules can be loaded or downloaded. This example dataset has 100
molecules.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Here, the dataset is downloaded. If you already have the</span></span>
<span><span class="co"># data locally, change URL to local path.</span></span>
<span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">sdfset</span> <span class="op">&lt;-</span> <span class="fu">read.SDFset</span><span class="op">(</span><span class="st">"http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/Samples/sdfsample.sdf"</span><span class="op">)</span></span>
<span>    <span class="co"># rename molecule IDs by IDs in the header. If your</span></span>
<span>    <span class="co"># molecules' header does not have ID or not unique,</span></span>
<span>    <span class="co"># remove following code and use the default IDs</span></span>
<span>    <span class="fu">cid</span><span class="op">(</span><span class="va">sdfset</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">makeUnique</span><span class="op">(</span><span class="fu">sdfid</span><span class="op">(</span><span class="va">sdfset</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"load_data"</span>, dependency <span class="op">=</span> <span class="st">"load_packages"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualize-molecules">Visualize molecules<a class="anchor" aria-label="anchor" href="#visualize-molecules"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"results/mols_plot.png"</span>, <span class="fl">700</span>, <span class="fl">600</span><span class="op">)</span></span>
<span>    <span class="co"># Here only first 4 are plotted. Please choose the ones</span></span>
<span>    <span class="co"># you want to plot.</span></span>
<span>    <span class="fu">ChemmineR</span><span class="fu">::</span><span class="fu">plot</span><span class="op">(</span><span class="va">sdfset</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"vis_mol"</span>, dependency <span class="op">=</span> <span class="st">"load_data"</span>, run_step <span class="op">=</span> <span class="st">"optional"</span><span class="op">)</span></span></code></pre></div>
<p><img src="results/mols_plot.png"></p>
</div>
<div class="section level3">
<h3 id="basic-molecule-inforamtion">Basic molecule inforamtion<a class="anchor" aria-label="anchor" href="#basic-molecule-inforamtion"></a>
</h3>
<p>Compute some basic molecule information and store to file, such as
atom frequency matrix, molecular weight and formula.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">propma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>MF <span class="op">=</span> <span class="fu">MF</span><span class="op">(</span><span class="va">sdfset</span><span class="op">)</span>, MW <span class="op">=</span> <span class="fu">MW</span><span class="op">(</span><span class="va">sdfset</span><span class="op">)</span>, <span class="fu">atomcountMA</span><span class="op">(</span><span class="va">sdfset</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_csv</a></span><span class="op">(</span><span class="va">propma</span>, <span class="st">"results/basic_mol_info.csv"</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"basic_mol_info"</span>, dependency <span class="op">=</span> <span class="st">"load_data"</span>, run_step <span class="op">=</span> <span class="st">"optional"</span><span class="op">)</span></span></code></pre></div>
<p>The information can be visualized, for example, a boxplot of atom
frequency.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"results/atom_req.png"</span>, <span class="fl">700</span>, <span class="fl">700</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">propma</span><span class="op">[</span>, <span class="fl">3</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">propma</span><span class="op">)</span><span class="op">]</span>, col <span class="op">=</span> <span class="st">"#6cabfa"</span>, main <span class="op">=</span> <span class="st">"Atom Frequency"</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"mol_info_plot"</span>, dependency <span class="op">=</span> <span class="st">"basic_mol_info"</span>,</span>
<span>    run_step <span class="op">=</span> <span class="st">"optional"</span><span class="op">)</span></span></code></pre></div>
<p><img src="results/atom_req.png"></p>
</div>
<div class="section level3">
<h3 id="convert-sdf-to-apfp">Convert SDF to AP/FP<a class="anchor" aria-label="anchor" href="#convert-sdf-to-apfp"></a>
</h3>
<p>Tanimoto similarity requires molecules to be converted into
searchable format. In this workflow, either atom pair (AP) or
fingerprint (FP) should be used as input. This step convert SDF
molecules to these formats.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">apset</span> <span class="op">&lt;-</span> <span class="fu">sdf2ap</span><span class="op">(</span><span class="va">sdfset</span><span class="op">)</span></span>
<span>    <span class="va">fpset</span> <span class="op">&lt;-</span> <span class="fu">desc2fp</span><span class="op">(</span><span class="va">apset</span>, descnames <span class="op">=</span> <span class="fl">1024</span>, type <span class="op">=</span> <span class="st">"FPset"</span><span class="op">)</span></span>
<span>    <span class="co"># save them to a file so they can be loaded directly</span></span>
<span>    <span class="co"># next time</span></span>
<span>    <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html" class="external-link">write_rds</a></span><span class="op">(</span><span class="va">apset</span>, <span class="st">"results/apset.rds"</span><span class="op">)</span></span>
<span>    <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html" class="external-link">write_rds</a></span><span class="op">(</span><span class="va">fpset</span>, <span class="st">"results/fpset.rds"</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"apfp_convert"</span>, dependency <span class="op">=</span> <span class="st">"load_data"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="remove-duplicates">Remove duplicates<a class="anchor" aria-label="anchor" href="#remove-duplicates"></a>
</h3>
<p>When molecules are converted to AP/FP, there is chance some very
similar molecules end up with idenitical AP/FP, for example some
stereoisomers. This is the shortness of these standardization methods.
For the downstream clustering part, we need to remove the AP/FP
duplicated ones. In this workflow, FPset is used for similarity and
clustering computation, FP deduplication is demonstrated here. It will
be very similar for AP.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">fpset</span> <span class="op">&lt;-</span> <span class="va">fpset</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="op">!</span><span class="fu">cmp.duplicated</span><span class="op">(</span><span class="va">apset</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"fp_dedup"</span>, dependency <span class="op">=</span> <span class="st">"apfp_convert"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="similarity-compute">Similarity compute<a class="anchor" aria-label="anchor" href="#similarity-compute"></a>
</h3>
<p>In this step, the pairwise similarity is computed.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="co"># If your set is very large, compute a small portion is</span></span>
<span>    <span class="co"># a better idea, e.g.  sapply(cid(fpset)[1:10], ...</span></span>
<span>    <span class="va">simMAfp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu">cid</span><span class="op">(</span><span class="va">fpset</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">fpSim</span><span class="op">(</span>x <span class="op">=</span> <span class="va">fpset</span><span class="op">[</span><span class="va">x</span><span class="op">]</span>,</span>
<span>        <span class="va">fpset</span>, sorted <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"fp_similarity"</span>, dependency <span class="op">=</span> <span class="st">"fp_dedup"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="hierarchical-clustering">Hierarchical clustering<a class="anchor" aria-label="anchor" href="#hierarchical-clustering"></a>
</h3>
<p>After the similarity computation, we can do some clustering. In this
step, hierarchical clustering (hclust) is performed.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="va">simMAfp</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"results/hclust.png"</span>, <span class="fl">1000</span>, <span class="fl">700</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="va">hc</span><span class="op">)</span>, edgePar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>col <span class="op">=</span> <span class="fl">4</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>        horiz <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="co"># to see the tree groupings, one need to cut the tree,</span></span>
<span>    <span class="co"># for example, by height of 0.9</span></span>
<span>    <span class="va">tree_cut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html" class="external-link">cutree</a></span><span class="op">(</span><span class="va">hc</span>, h <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span></span>
<span>    <span class="va">grouping</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>cid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tree_cut</span><span class="op">)</span>, group_id <span class="op">=</span> <span class="va">tree_cut</span><span class="op">)</span></span>
<span>    <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_csv</a></span><span class="op">(</span><span class="va">grouping</span>, <span class="st">"results/hclust_grouping.csv"</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"hclust"</span>, dependency <span class="op">=</span> <span class="st">"fp_similarity"</span>, run_step <span class="op">=</span> <span class="st">"optional"</span><span class="op">)</span></span></code></pre></div>
<p><img src="results/hclust.png"></p>
<p>With the group information, directly plotting the hclust does not
look nice. We can use other methods to visulize the grouping. Here, PCA
is used.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span>    <span class="co"># simMAfp is already 0-1 value, no need to normalize</span></span>
<span>    <span class="co"># again</span></span>
<span>    <span class="va">mol_pca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/princomp.html" class="external-link">princomp</a></span><span class="op">(</span><span class="va">simMAfp</span><span class="op">)</span></span>
<span>    <span class="co"># find the variance</span></span>
<span>    <span class="va">mol_pca_var</span> <span class="op">&lt;-</span> <span class="va">mol_pca</span><span class="op">$</span><span class="va">sdev</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">^</span><span class="fl">2</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">mol_pca</span><span class="op">$</span><span class="va">sdev</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span></span>
<span>    <span class="co"># plot</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"results/mol_pca.png"</span>, <span class="fl">650</span>, <span class="fl">700</span><span class="op">)</span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>PC1 <span class="op">=</span> <span class="va">mol_pca</span><span class="op">$</span><span class="va">loadings</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, PC2 <span class="op">=</span> <span class="va">mol_pca</span><span class="op">$</span><span class="va">loadings</span><span class="op">[</span>,</span>
<span>        <span class="fl">2</span><span class="op">]</span>, group_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">grouping</span><span class="op">$</span><span class="va">group_id</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="co"># here only group labels are used to color, if you</span></span>
<span>        <span class="co"># have experimental conditions or drug types, the</span></span>
<span>        <span class="co"># coloring or shaping will be more meaningful.</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, color <span class="op">=</span> <span class="va">group_id</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ellipse.html" class="external-link">stat_ellipse</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC1 "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">mol_pca_var</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>        <span class="fl">3</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>, <span class="st">"%"</span><span class="op">)</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC1 "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">mol_pca_var</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>        <span class="fl">3</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>, <span class="st">"%"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/theme_pubr.html" class="external-link">theme_pubr</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set2"</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"PCA"</span>, dependency <span class="op">=</span> <span class="st">"hclust"</span>, run_step <span class="op">=</span> <span class="st">"optional"</span><span class="op">)</span></span></code></pre></div>
<p><img src="results/mol_pca.png"></p>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>Alternatively, we can plot a heatmap to see the similarities between
molecules.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/talgalili/gplots" class="external-link">gplots</a></span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"results/mol_heatmap.png"</span>, <span class="fl">700</span>, <span class="fl">700</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/gplots/man/heatmap.2.html" class="external-link">heatmap.2</a></span><span class="op">(</span><span class="va">simMAfp</span>, Rowv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="va">hc</span><span class="op">)</span>, Colv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="va">hc</span><span class="op">)</span>,</span>
<span>        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/gplots/man/colorpanel.html" class="external-link">colorpanel</a></span><span class="op">(</span><span class="fl">40</span>, <span class="st">"darkblue"</span>, <span class="st">"yellow"</span>, <span class="st">"white"</span><span class="op">)</span>,</span>
<span>        density.info <span class="op">=</span> <span class="st">"none"</span>, trace <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"heatmap"</span>, dependency <span class="op">=</span> <span class="st">"fp_similarity"</span>, run_step <span class="op">=</span> <span class="st">"optional"</span><span class="op">)</span></span></code></pre></div>
<p><img src="results/mol_heatmap.png"></p>
</div>
<div class="section level3">
<h3 id="workflow-session">Workflow session<a class="anchor" aria-label="anchor" href="#workflow-session"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"wf_session"</span>, dependency <span class="op">=</span> <span class="st">"heatmap"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="manage-the-workflow">Manage the workflow<a class="anchor" aria-label="anchor" href="#manage-the-workflow"></a>
</h2>
<p>For running the workflow, <code>runWF</code> function will execute
all the steps store in the workflow container. The execution will be on
a single machine without submitting to a queuing system of a computer
cluster.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runWF.html" class="external-link">runWF</a></span><span class="op">(</span><span class="va">sal</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>To use complex workflow control options, such as parallelization,
subsetting samples, selecting steps, read the <a href="https://systempipe.org/sp/spr/sp_run/step_run/" class="external-link">documents</a> on
our website.</li>
<li>Explore <a href="https://systempipe.org/sp/spr/sp_run/sal_explore/" class="external-link">other details
of the workflow object</a>.</li>
<li>Create <a href="https://systempipe.org/sp/spr/sp_run/step_reports/" class="external-link">logs and
reports</a>.</li>
<li>
<a href="https://systempipe.org/sp/spr/sp_run/step_vis/" class="external-link">Visualize
the workflow</a>.</li>
</ul>
</div>
<div class="section level2">
<h2 id="about-the-workflow">About the workflow<a class="anchor" aria-label="anchor" href="#about-the-workflow"></a>
</h2>
<div class="section level3">
<h3 id="tools-used">Tools used<a class="anchor" aria-label="anchor" href="#tools-used"></a>
</h3>
<p>To check command-line tools used in this workflow, use
<code>listCmdTools</code>, and use <code>listCmdModules</code> to check
if you have a modular system.</p>
<p>The following code will print out tools required in your custom SPR
project in the report. In case you are running the workflow for the
first and do not have a project yet, or you just want to browser this
workflow, following code displays the tools required by default.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">".SPRproject"</span>, <span class="st">"SYSargsList.yml"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/eval.html" class="external-link">local</a></span><span class="op">(</span><span class="op">{</span></span>
<span>        <span class="va">sal</span> <span class="op">&lt;-</span> <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SPRproject.html" class="external-link">SPRproject</a></span><span class="op">(</span>resume <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>        <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/listCmdTools.html" class="external-link">listCmdTools</a></span><span class="op">(</span><span class="va">sal</span><span class="op">)</span></span>
<span>        <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/listCmdTools.html" class="external-link">listCmdModules</a></span><span class="op">(</span><span class="va">sal</span><span class="op">)</span></span>
<span>    <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu">crayon</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/crayon/man/crayon.html" class="external-link">blue</a></span><span class="op">$</span><span class="fu">bold</span><span class="op">(</span><span class="st">"Tools and modules required by this workflow are:\n"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"No other tool is required"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #0000BB; font-weight: bold;">Tools and modules required by this workflow are:</span></span></span>
<span><span class="co"><span style="color: #0000BB; font-weight: bold;">## </span>No other tool is required</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<p>This is the session information for rendering this report. To access
the session information of workflow running, check HTML report of
<code>renderLogs</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/Los_Angeles</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [6] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] BiocStyle_2.28.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] crayon_1.5.2          vctrs_0.6.2          </span></span>
<span><span class="co">##  [3] cli_3.6.1             knitr_1.42           </span></span>
<span><span class="co">##  [5] rlang_1.1.1           xfun_0.39            </span></span>
<span><span class="co">##  [7] stringi_1.7.12        purrr_1.0.1          </span></span>
<span><span class="co">##  [9] textshaping_0.3.6     jsonlite_1.8.4       </span></span>
<span><span class="co">## [11] glue_1.6.2            rprojroot_2.0.3      </span></span>
<span><span class="co">## [13] htmltools_0.5.5       formatR_1.14         </span></span>
<span><span class="co">## [15] ragg_1.2.5            sass_0.4.5           </span></span>
<span><span class="co">## [17] rmarkdown_2.21        evaluate_0.20        </span></span>
<span><span class="co">## [19] jquerylib_0.1.4       fastmap_1.1.1        </span></span>
<span><span class="co">## [21] lifecycle_1.0.3       yaml_2.3.7           </span></span>
<span><span class="co">## [23] memoise_2.0.1         bookdown_0.33        </span></span>
<span><span class="co">## [25] BiocManager_1.30.21.1 stringr_1.5.0        </span></span>
<span><span class="co">## [27] compiler_4.3.1        codetools_0.2-19     </span></span>
<span><span class="co">## [29] fs_1.6.2              rstudioapi_0.14      </span></span>
<span><span class="co">## [31] systemfonts_1.0.4     digest_0.6.31        </span></span>
<span><span class="co">## [33] R6_2.5.1              magrittr_2.0.3       </span></span>
<span><span class="co">## [35] bslib_0.4.2           tools_4.3.1          </span></span>
<span><span class="co">## [37] pkgdown_2.0.7         cachem_1.0.8         </span></span>
<span><span class="co">## [39] desc_1.4.2</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Le Zhang, <a href="http://girke.bioinformatics.ucr.edu/" class="external-link">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
